src/graph/resolver.rs

```diff
@@ -242,81 +251,101 @@ impl Resolver {
                 return false;
             }
 
-            // Compare individual color attachments for compatibility
-            for (attachment_idx, lhs_attachment) in lhs_exec
-                .color_attachments
-                .iter()
-                .chain(lhs_exec.color_loads.iter())
-                .chain(lhs_exec.color_stores.iter())
-                .chain(
-                    lhs_exec
-                        .color_clears
-                        .iter()
-                        .map(|(attachment_idx, (attachment, _))| (attachment_idx, attachment)),
-                )
-                .chain(
-                    lhs_exec
-                        .color_resolves
-                        .iter()
-                        .map(|(attachment_idx, (attachment, _))| (attachment_idx, attachment)),
-                )
-            {
-                let rhs_attachment = rhs_exec
+            // Check attachment compatibility both ways since one may have a different quantity than the other
```

Comment:
Note: I don't think we need to check both ways here. Just verify the lengths are the same and then check one way like normal.

Reply:
I think this check is valid, but the way it is implemented is less than optimal. Assuming the attachments below are compatible where specified:

Pass A | Pass B | Result
-- | -- | --
Color0 + Depth | Depth | Merge
Depth | Color0 + Depth | Merge
Color0 | Depth | Don't Merge

The first row is the `(LHS, RHS)` check and the second row is the `(RHS, LHS)` check, just checking the length would cause them to not merge.

I think a much better way to do this check would be to step through both attachment lists at the same time, so that all indexes are visited just once. This is hard because of the way `Execution` stores so many individual hashmaps for attachments, and resolving that would make improving this function easier.

Reply:
Oh interesting! I actually didn't know you could merge `Color0 + Depth` and `Depth`